src/colorizer/CanvasOverlay.ts

```diff
@@ -90,47 +191,202 @@ export default class CanvasOverlay {
+    const useSeconds = !(timestampOptions.frameDurationSec % 60 === 0 && timestampOptions.startTimeSec % 60 === 0);
+
+    const timestampDigits: string[] = [];
+    const timestampUnits: string[] = [];
+
+    if (useHours) {
+      const hours = Math.floor(timestampOptions.currTimeSec / (60 * 60));
+      timestampDigits.push(hours.toString().padStart(2, "0"));
+      timestampUnits.push("h");
+    }
+    if (useMinutes) {
+      const minutes = Math.floor(timestampOptions.currTimeSec / 60) % 60;
+      timestampDigits.push(minutes.toString().padStart(2, "0"));
+      timestampUnits.push("m");
+    }
+    if (useSeconds) {
+      const seconds = timestampOptions.currTimeSec % 60;
+      if (!useHours && timestampOptions.frameDurationSec % 1.0 !== 0) {
+        // Duration increment is smaller than a second and we're not showing hours, so show milliseconds.
+        timestampDigits.push(seconds.toFixed(3).padStart(6, "0"));
+      } else {
+        timestampDigits.push(seconds.toFixed(0).padStart(2, "0"));
+      }
+      timestampUnits.push("s");
+    }
+
+    return timestampDigits.join(":") + " (" + timestampUnits.join(", ") + ")";
+  }
+
+  /**
+   * Draws the timestamp, if visible, and returns the rendered height and width.
+   * @param originPx The origin of the timestamp, from the lower right corner, in pixels.
+   * @returns an object with two properties:
+   *  - `size`: a vector representing the width and height of the rendered scale bar, in pixels.
+   *  - `render`: a callback that renders the scale bar to the canvas.
+   */
+  private renderTimestamp(ctx: CanvasRenderingContext2D, originPx: Vector2): RenderInfo {
+    if (!this.timestampOptions.visible) {
+      return { sizePx: new Vector2(0, 0), render: () => {} };
+    }
+
+    ////////////////// Format timestamp as text //////////////////
+    const timestampFormatted = CanvasOverlay.getTimestampLabel(this.timestampOptions);
+
+    // TODO: Would be nice to configure top/bottom/left/right padding separately.
+    const timestampPaddingPx = new Vector2(6, 2);
+    const timestampOriginPx = new Vector2(originPx.x + timestampPaddingPx.x, originPx.y + timestampPaddingPx.y);
+    // Save the render function for later.
+    const render = (): void => {
+      this.renderRightAlignedText(ctx, timestampOriginPx, timestampFormatted, this.scaleBarOptions);
+    };
+
+    return {
+      sizePx: new Vector2(
+        timestampPaddingPx.x * 2 + this.getTextDimensions(ctx, timestampFormatted, this.timestampOptions).x,
+        timestampPaddingPx.y * 2 + this.timestampOptions.fontSizePx
+      ),
+      render,
+    };
+  }
+
+  /**
+   * Draws the background overlay in the bottom right corner of the canvas.
+   * @param ctx Canvas context to render to.
+   * @param size Size of the background overlay.
+   * @param options Configuration for the background overlay.
+   */
+  private static renderBackground(ctx: CanvasRenderingContext2D, size: Vector2, options: BackgroundOptions): void {
```

Comment:
minor concern for future: what if a user said they want the time overlay at bottom left and scale bar at bottom right?  Are they too tightly coupled in this code, just for the sake of having one single background rectangle?

Reply:
Super fair! One of the nice things about the current code is that, because each element reports its dimensions, they're not inherently dependent on each other for rendering, and don't make assumptions about other elements existing.

But yeah, all the current components assume they're in the same corner for now. The styling and options objects will make it easier to configure this in the future.